File: 0409-longest-palindrome/0409-longest-palindrome.cs
public class Solution {
    public int LongestPalindrome(string s) 
    {
        IDictionary<char, int> counts = new Dictionary<char, int>();
        
        foreach (char c in s)
        {
            if (!counts.ContainsKey(c))
            {
                counts.Add(c, 0);
            }
            counts[c]++;
        }
        
        bool flag = false;
        int res = 0;
        
        foreach (var entry in counts)
        {
            if (entry.Value % 2 == 0)
            {
                res += entry.Value;
            }
            else 
            {
                flag = true;
                res += (entry.Value - 1);
            }
        }
        
        return flag ? res + 1 : res;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
const longestPalindrome = function(s) {
    const counts = {};
    for (const char of s) {
        counts[char] ||= 0; 
        counts[char]++;
    };
    let sum = 0;
    let flag = false;
    for (const count of Object.values(counts)) {
        if (count % 2 === 0) sum += count;
        else {
            sum += (count - 1);
            flag = true;
        }
    }
    return flag ? ++sum : sum;
}